plugins/modules/ldap_inc.py

```diff
@@ -0,0 +1,236 @@
+        required_together=ldap_required_together(),
+    )
+
+    deps.validate(module)
+
+    # Instantiate the LdapAttr object
+    mod = LdapInc(module)
+
+    changed = False
+    ret = ""
+    rfc4525 = False
+
+    try:
+        if mod.increment != 0:
+            changed = True
+
+            if not module.check_mode:
+
+                if mod.method != "auto":
+                    rfc4525 = mod.method == "rfc425"
+                else:
+                    rootDSE = mod.connection.search_ext_s(
+                        base="",
+                        scope=ldap.SCOPE_BASE,
+                        attrlist=["*", "+"])
+                    if len(rootDSE) == 1:
+                        if to_bytes(ldap.CONTROL_POST_READ) in rootDSE[0][1]["supportedControl"] and (
+                            mod.LDAP_MOD_INCREMENT in rootDSE[0][1]["supportedFeatures"] or
+                            mod.LDAP_MOD_INCREMENT in rootDSE[0][1]["supportedExtension"]
+                        ):
+                            rfc4525 = True
+
+                if rfc4525:
+                    dummy, dummy, dummy, resp_ctrls = mod.connection.modify_ext_s(
+                        dn=mod.dn,
+                        modlist=mod.inc_rfc4525(),
+                        serverctrls=mod.serverControls(),
+                        clientctrls=None)
+                    if len(resp_ctrls) == 1:
+                        ret = resp_ctrls[0].entry[mod.attr][0]
+
+                else:
+                    tries = 0
+                    max_tries = 3
+                    while tries < max_tries:
+                        tries = tries + 1
+                        result = mod.connection.search_ext_s(
+                            base=mod.dn,
+                            scope=ldap.SCOPE_BASE,
+                            filterstr="(%s=*)" % mod.attr,
+                            attrlist=[mod.attr])
+                        if len(result) != 1:
+                            module.fail_json(msg="The entry does not exist or does not contain the specified attribute.")
+                            return
+                        try:
+                            ret = str(int(result[0][1][mod.attr][0]) + mod.increment)
+                            # if the current value first arg in inc_legacy has changed then the modify will fail
+                            mod.connection.modify_s(
+                                dn=mod.dn,
+                                modlist=mod.inc_legacy(result[0][1][mod.attr][0], ret))
+                            break
+                        except ldap.NO_SUCH_ATTRIBUTE:
+                            if tries == max_tries:
+                                module.fail_json(msg="The increment could not be applied after " + str(max_tries) + " tries.")
+                                return
+                        except Exception as e0:
+                            raise e0
+
+        else:
+            if not module.check_mode:
```

Comment:
Is this check needed? From what I understand the next lines do not modify anything, but just return the value.

Reply:
This part is reviewed.